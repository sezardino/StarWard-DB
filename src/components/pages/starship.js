import React, {PureComponent} from 'react';
import Row from '../row';
import {StarshipsList, StarshipDetails} from '../app-components';

export default class StarshipPage extends PureComponent {
	state = {
		selectedItem: null,
	};

	itemClickHandler = (selectedItem) => {
		this.setState({selectedItem});
	};

	render() {
		const {selectedItem} = this.state;
		return (
			<Row
				left={
					<StarshipsList
						renderItem={({name}) => `${name}`}
						onListItemClick={this.itemClickHandler}
					/>
				}
				right={<StarshipDetails dataId={selectedItem} />}
			/>
		);
	}
}
