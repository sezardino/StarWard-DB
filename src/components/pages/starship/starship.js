import React from 'react';
import {withRouter} from 'react-router-dom';
import {StarshipsList} from '../../app-components';

const StarshipPage = ({history}) => {
	return (
		<StarshipsList
			renderItem={({name}) => `${name}`}
			onListItemClick={(itemId) => history.push(itemId)}
		/>
	);
};

export {StarshipPage};
export default withRouter(StarshipPage);
