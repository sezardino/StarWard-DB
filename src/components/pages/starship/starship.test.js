import React from 'react';
import {shallow} from 'enzyme';

import {StarshipPage} from './starship';

describe('Component StarshipPage', () => {
	const mockMatch = {params: {id: 'Test id'}};
	const component = shallow(<StarshipPage match={mockMatch} />);
	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});
});
