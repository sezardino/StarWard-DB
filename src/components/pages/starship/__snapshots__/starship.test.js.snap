// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StarshipPage Component rendered correctly 1`] = `"<Component renderItem={[Function: renderItem]} onListItemClick={[Function: onListItemClick]} />"`;
